src/Microsoft.DotNet.Darc/src/DarcLib/VirtualMonoRepo/VmrPusher.cs

```diff
@@ -59,17 +59,15 @@ public async Task Push(string remote, string branch, bool verifyCommits, string?
         }  
     }
 
-    private void ExecutePush(
-        string remoteName,
-        string branchName)
+    private void ExecutePush(string remoteName, string branchName)
     {
         using var vmrRepo = new Repository(_vmrInfo.VmrPath);
         vmrRepo.Config.Add("user.name", Constants.DarcBotName);
         vmrRepo.Config.Add("user.email", Constants.DarcBotEmail);
 
         if(vmrRepo.Network.Remotes.FirstOrDefault(r => r.Name == remoteName) == null)
         {
-            throw new Exception($"No remote with name {remoteName} found in VMR repo.");
+            throw new Exception($"No remote named {remoteName} found in VMR repo.");
         }
 
         var remote = vmrRepo.Network.Remotes[remoteName];
```

Comment:
Do not perform a double look-up. Assign `FirstOrDefault` result into a variable right away and compare for `null`

Reply:
Ideally check whether there is something like `TryGetValue` on the collection. Not sure there is though

Reply:
no, it's some custom enumerable collection

Reply:
actually the [] operators on `Remotes` and `Branches` have some more logic that just searching by name so I think it's better to use those

Reply:
no, it returns null